src/hotspot/share/c1/c1_Runtime1.cpp

```diff
@@ -1269,6 +1269,8 @@ JRT_END
 void Runtime1::patch_code(JavaThread* thread, Runtime1::StubID stub_id) {
   NOT_PRODUCT(_patch_code_slowcase_cnt++);
 
+  Thread::WXWriteFromExecSetter wx_write;
```

Comment:
Could you please elaborate what this addition fixes ?
Maybe we have missed something in our patch for jit

Reply:
`DEOPTIMIZE_WHEN_PATCHING` is defined in `src/hotspot/cpu/aarch64/globalDefinitions_aarch64.hpp`. The crash I am observing is a SIGBUS with the following stack trace:
```
Stack: [0x000000016b14c000,0x000000016b34f000],  sp=0x000000016b349ce0,  free space=2039k
Native frames: (J=compiled Java code, A=aot compiled Java code, j=interpreted, Vv=VM code, C=native code)
V  [libjvm.dylib+0x22b6e0]  int Atomic::PlatformAdd<4ul>::add_and_fetch<int, int>(int volatile*, int, atomic_memory_order) const+0x2c
V  [libjvm.dylib+0x22b6a8]  Atomic::AddImpl<int, int, void>::add_and_fetch(int volatile*, int, atomic_memory_order)+0x34
V  [libjvm.dylib+0x22b668]  int Atomic::add<int, int>(int volatile*, int, atomic_memory_order)+0x28
V  [libjvm.dylib+0x224084]  void Atomic::inc<int>(int volatile*, atomic_memory_order)+0x24
V  [libjvm.dylib+0xad33b4]  nmethodLocker::lock_nmethod(CompiledMethod*, bool)+0x60
V  [libjvm.dylib+0xad7fac]  nmethodLocker::nmethodLocker(nmethod*)+0x5c
V  [libjvm.dylib+0xad0450]  nmethodLocker::nmethodLocker(nmethod*)+0x28
V  [libjvm.dylib+0xacff98]  nmethod::make_not_entrant_or_zombie(int)+0xa0
V  [libjvm.dylib+0xad7504]  nmethod::make_not_entrant()+0x5c
V  [libjvm.dylib+0x331114]  Runtime1::patch_code(JavaThread*, Runtime1::StubID)+0x108
V  [libjvm.dylib+0x33123c]  Runtime1::move_mirror_patching(JavaThread*)+0x48
v  ~RuntimeStub::load_mirror_patching Runtime1 stub
C  0x000000016b34b2b8
```

Reply:
Hm, it's interesting how one can trigger that ?
I can pass tier1/tier2 without this patch just fine, makes me wonder if we forgot something in our code drop.

Reply:
I trigger it with a simple `java -Xcomp -version`